### PR TITLE
Adds Option to Copy Panel to Clipboard

### PR DESCRIPTION
**Pull request by Animeboynz**
Uses Android's ClipboardManager to copy an Panel to the clipboard.

### CODE REVIEW

**Comment by Animeboynz** [comment]
Damb you Build Checks.

**Comment by Animeboynz** [comment]
Well finally down to one issue and its one I'm definitely not going to attempt as I'll probably break the entire thing xD

`/home/runner/work/mihon/mihon/app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderViewModel.kt:84:7: Class ReaderViewModel is too large. Consider splitting it into smaller pieces. [LargeClass]`

**Comment by Animeboynz** [comment]
@AntsyLich This is basically done from my end, just let me know if you need me to remove the UI or if there is anything you need me to do

**Comment by AntsyLich** [comment]
Rebase and fix build thanks

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderViewModel.kt [hunk 1]
```diff
@@ -809,7 +809,8 @@ class ReaderViewModel @JvmOverloads constructor(
      * get a path to the file and it has to be decompressed somewhere first. Only the last shared
      * image will be kept so it won't be taking lots of internal disk space.
      */
-    fun shareImage() {
+
+    private fun handleImage(isCopy: Boolean) {
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderViewModel.kt:813]:
```suggestion
    fun shareImage(copy: Boolean) {
```

**Review by AntsyLich — CHANGES_REQUESTED** [review]
Also do a rebase instead of merging. It's a total mess

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderViewModel.kt (hunk 1 above)
Comment (AntsyLich) [reply app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderViewModel.kt:813]: or
```kt
    fun shareImage(copyToClipboard: Boolean) {
```

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderActivity.kt [hunk 2]
```diff
@@ -721,6 +726,13 @@ class ReaderActivity : BaseActivity() {
         startActivity(Intent.createChooser(intent, stringResource(MR.strings.action_share)))
     }
 
+    private fun onCopyImageResult(uri: Uri) {
+        // Copy the URI to the clipboard
+        val clipboardManager = applicationContext.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderActivity.kt:731]: will require an import
```suggestion
        val clipboardManager = applicationContext.getSystemService<ClipboardManager>()
```

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderActivity.kt [hunk 3]
```diff
@@ -721,6 +727,13 @@ class ReaderActivity : BaseActivity() {
         startActivity(Intent.createChooser(intent, stringResource(MR.strings.action_share)))
     }
 
+    private fun onCopyImageResult(uri: Uri) {
+        // Copy the URI to the clipboard
+        val clipboardManager = applicationContext.getSystemService<ClipboardManager>()
+        val clipData = ClipData.newUri(applicationContext.contentResolver, "Image URI", uri)
+        clipboardManager?.setPrimaryClip(clipData)
+    }
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderActivity.kt:735]:
```suggestion
    private fun onCopyImageResult(uri: Uri) {
        val clipboardManager = applicationContext.getSystemService<ClipboardManager>() ?: return
        val clipData = ClipData.newUri(applicationContext.contentResolver, "", uri)
        clipboardManager.setPrimaryClip(clipData)
    }
```

**Pull request merged**